Simulation/Derivatives.m

```diff
@@ -24,6 +24,12 @@
 
   kOut.V_dot = Gravity + F / SimData.Plane.Mass; % Derivative of velocity is acceleration
 
+  % Do not apply downward acceleration or velocity when on ground
+  if SimData.Plane.FSM_state == 0
+    kOut.P_dot(2) = 0;
+    kOut.V_dot(2) = 0;
```

Comment:
These should not be zeroed out, they should be minimized to 0, which would allow for positive acceleration and velocity from landed state

Reply:
I'm not sure what you mean by "minimized" but I assume you want me to add another check here to make sure if velocity and acceleration are actually below 0. That is an oversight on my part.

Reply:
As in ` kOut.P_dot(2) = min(0,kOut.P_dot(2));` so that if it's positive it won't be affected but if it's negative it won't cause the plane to go below the ground

Reply:
Ah yeah, that is the same thing effectively. I feel like adding it in as a condition of the if statement would be a bit neater.

Reply:
Sounds good to me, I'll leave it up to you!